p2p/node.go

```diff
@@ -454,3 +530,57 @@ func (n *Node) receive(ctx context.Context) (*Message, error) {
 	}
 	return &Message{From: msg.GetFrom(), Data: msg.Data}, nil
 }
+
+func ipNetFromMaddr(maddr ma.Multiaddr) (ipNet net.IPNet, err error) {
+	ip, err := ipFromMaddr(maddr)
+	if err != nil {
+		return net.IPNet{}, err
+	}
+	mask := getAllMaskForIP(ip)
+	return net.IPNet{
+		IP:   ip,
+		Mask: mask,
+	}, nil
+}
+
+func ipFromMaddr(maddr ma.Multiaddr) (net.IP, error) {
+	var (
+		ip    net.IP
+		found bool
+	)
+
+	ma.ForEach(maddr, func(c ma.Component) bool {
+		switch c.Protocol().Code {
+		case ma.P_IP6ZONE:
+			return true
+		case ma.P_IP6, ma.P_IP4:
+			found = true
+			ip = net.IP(c.RawValue())
+			return false
+		default:
+			return false
```

Comment:
What does returning `true` vs. `false` do here?

Reply:
Returning false causes `ma.ForEach` to stop iterating through components of the multiaddress. Returning true will cause it to keep iterating.